internal/controlplane/handlers_authz.go

```diff
@@ -567,29 +572,22 @@ func (s *Server) removeRole(
 	subject string,
 ) (*minder.RemoveRoleResponse, error) {
 	var err error
-	// We may be given a human-readable identifier which can vary over time. Resolve
-	// it to an IDP-specific stable identifier so that we can support subject renames.
-	identity := &auth.Identity{
-		Provider:  nil,
-		UserID:    subject,
-		HumanName: subject,
-	}
-	if flags.Bool(ctx, s.featureFlags, flags.IDPResolver) {
-		identity, err = s.idClient.Resolve(ctx, subject)
-		if err != nil {
-			zerolog.Ctx(ctx).Error().Err(err).Msg("error resolving identity")
-			return nil, util.UserVisibleError(codes.NotFound, "could not find identity %q", subject)
-		}
+	// Resolve the subject to an identity
+	identity, err := s.idClient.Resolve(ctx, subject)
+	if err != nil {
+		zerolog.Ctx(ctx).Error().Err(err).Msg("error resolving identity")
+		return nil, util.UserVisibleError(codes.NotFound, "could not find identity %q", subject)
 	}
 
 	// Verify if user exists
-	if _, err := s.store.GetUserBySubject(ctx, identity.UserID); err != nil {
+	if _, err := s.store.GetUserBySubject(ctx, identity.String()); err != nil {
 		if errors.Is(err, sql.ErrNoRows) {
 			return nil, util.UserVisibleError(codes.NotFound, "User not found")
 		}
 		return nil, status.Errorf(codes.Internal, "error getting user: %v", err)
 	}
 
+	// Delete the role assignment
 	if err := s.authzClient.Delete(ctx, identity.String(), role, targetProject); err != nil {
 		return nil, status.Errorf(codes.Internal, "error writing role assignment: %v", err)
 	}
```

Comment:
Do we want to allow a user to remove themselves from the project?

Reply:
We probably also want to enforce the invariant that projects without parents have at least one admin after `RemoveRole`.

Reply:
I'll ask @ethomson for the first one. And I agree on the 2nd one, I'll add it to the PR with the rest of the fixes 👍

Reply:
Ed confirmed on the 1st one 👍 I'll add a follow up.